samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -145,10 +145,14 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
     * Gets the physical name for the specified streamId.
     *
     * @param streamId             the identifier for the stream in the config.
-    * @param defaultPhysicalName  the default to use if the physical name is missing.
     * @return                     the physical identifier for the stream or the default if it is undefined.
     */
-  def getPhysicalName(streamId: String, defaultPhysicalName: String) = {
+  def getPhysicalName(streamId: String) = {
+    // Add prefix to default so the physical name is unique per job
+    val defaultPhysicalName = String.format("%s-%s-%s",
```

Comment:
This is a great point. I think by adding the prefix as default physical name does violate the expectation that streamId is physicalId in normal cases. I refactored it so the prefix is only appended in the logical ID. Thanks a lot for catching this!